docker/build.sh

```diff
@@ -1,54 +1,171 @@
-    dev=""
-elif [ "$#" == 2 ]; then
-    if [[ "$2" != "--dev" && "$2" != "dev" ]]; then
-        echo "ERROR: argument 2 needs to be \`dev\` or \`--dev\`, not \`$2\`"
-        exit 1
-    fi
-    dev="-dev"
-else
-    echo "Usage: bash docker/build.sh <framework> [dev]"
-    echo "    <framework> be \`tf1\`, \`tf2\`, \`pytorch\`, \`pytorch-deepspeech\`, or \`all\`"
-    exit 1
+
+Help()
+{
+  # Display Help
+  echo "----------------------------------------------------------------------------------------"
+  echo "                   Armory Docker Build Script                       "
+  echo
+  echo "This build script helps to build the docker images necessary for armory exectuion "
+  echo "(in docker mode).  The primary purpose of this script is to build the images from "
+  echo "scratch and to be used by the armory CI toolchain.  If modifications are necessary"
+  echo "you can use \`--dry-run\` to get the \`docker build...\` commands directly and "
+  echo "modify them as necessary. "
+  echo ""
+  echo "Armory uses scm versioning from git, therefore if you want to re-build containers "
+  echo "from a stable armory release, simply checkout that tag and then run this script"
+  echo ""
+  echo "Syntax: build.sh [-f|nc|dr|v|h]"
+  echo ""
+  echo "REQUIRED"
+  echo "-f | --framework            Select Which framework to target:"
+  echo "                                [all|base|tf2|pytorch|pytorch-deepspeech]"
+  echo ""
+  echo "OPTIONAL"
+  echo "-t | --tag                  Specify Additional Tag to apply to each image"
+  echo "-nc | --no-cache            Build Images \`Clean\` (i.e. using --no-cache)"
+  echo "-dr | --dry-run             Only show the Build calls (do not execute the builds)"
+  echo "-v | --verbose              Show Logs in Plain Text (uses \`--progress=plain\`"
+  echo "-h | --help                 Print this Help."
+  echo
+  echo "----------------------------------------------------------------------------------------"
+}
+
+get_tag_from_version ()
+{
+  local  input=$1
+  arr=( ${input//\./ } ) # Split by .
+  arr="${arr[@]:0:4}" # Keep 1st 4 elements
+  arr="${arr// /.}" # Put it back together
+  result="${arr/+/-}" # Replace + with -
+  echo $result
+}
+
+# Setting Defaults
+POSITIONAL_ARGS=()
+NO_CACHE=false
+ARMORY_VERSION="$(python setup.py --version)"
+DRYRUN=false
+VERBOSE="--progress=auto"
+REPO="twosixarmory"
+FRAMEWORK=""
+TAG=$(get_tag_from_version $ARMORY_VERSION)
+ADDITIONAL_TAG=false
+
+
+# Making sure execution path is correct
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+if [ "$PWD" != "$(dirname $SCRIPT_DIR)" ]; then
+  echo "Must Execute build script from within armory/ folder that contains folder called \`docker\`"
+  return 1
 fi
 
+## Parsing CLI Arguments
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -f|--framework)
+      FRAMEWORK="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    -t|--tag)
+      ADDITIONAL_TAG="$2"
```

Comment:
This needs better documentation because I don't know what to set it to and if un-specified yields `repo/name:false` as the docker name.